sph/include/sph/cooling.hpp

```diff
@@ -0,0 +1,141 @@
+    gr_float z_velocity                    = 0.;
+    grackle_fields.z_velocity              = &z_velocity;
+    gr_float HI_density                    = (gr_float)d.HI_fraction[i] * (gr_float)d.rho[i];
+    gr_float HII_density                   = (gr_float)d.HII_fraction[i] * (gr_float)d.rho[i];
+    gr_float HeI_density                   = (gr_float)d.HeI_fraction[i] * (gr_float)d.rho[i];
+    gr_float HeII_density                  = (gr_float)d.HeII_fraction[i] * (gr_float)d.rho[i];
+    gr_float HeIII_density                 = (gr_float)d.HeIII_fraction[i] * (gr_float)d.rho[i];
+    gr_float e_density                     = (gr_float)d.e_fraction[i] * (gr_float)d.rho[i];
+    gr_float HM_density                    = (gr_float)d.HM_fraction[i] * (gr_float)d.rho[i];
+    gr_float H2I_density                   = (gr_float)d.H2I_fraction[i] * (gr_float)d.rho[i];
+    gr_float H2II_density                  = (gr_float)d.H2II_fraction[i] * (gr_float)d.rho[i];
+    gr_float DI_density                    = (gr_float)d.DI_fraction[i] * (gr_float)d.rho[i];
+    gr_float DII_density                   = (gr_float)d.DII_fraction[i] * (gr_float)d.rho[i];
+    gr_float HDI_density                   = (gr_float)d.HDI_fraction[i] * (gr_float)d.rho[i];
+    gr_float metal_density                 = (gr_float)d.metal_fraction[i] * (gr_float)d.rho[i];
+    grackle_fields.HI_density              = &HI_density;
+    grackle_fields.HII_density             = &HII_density;
+    grackle_fields.HeI_density             = &HeI_density;
+    grackle_fields.HeII_density            = &HeII_density;
+    grackle_fields.HeIII_density           = &HeIII_density;
+    grackle_fields.e_density               = &e_density;
+    grackle_fields.HM_density              = &HM_density;
+    grackle_fields.H2I_density             = &H2I_density;
+    grackle_fields.H2II_density            = &H2II_density;
+    grackle_fields.DI_density              = &DI_density;
+    grackle_fields.DII_density             = &DII_density;
+    grackle_fields.HDI_density             = &HDI_density;
+    grackle_fields.metal_density           = &metal_density;
+    gr_float vhr                           = 0.;
+    gr_float shr                           = 0.;
+    gr_float HI_ir                         = 0.;
+    gr_float HeI_ir                        = 0.;
+    gr_float HeII_ir                       = 0.;
+    gr_float H2_dr                         = 0.;
+    gr_float hr                            = 0.;
+    grackle_fields.volumetric_heating_rate = &vhr;
+    grackle_fields.specific_heating_rate   = &shr;
+    grackle_fields.RT_HI_ionization_rate   = &HI_ir;
+    grackle_fields.RT_HeI_ionization_rate  = &HeI_ir;
+    grackle_fields.RT_HeII_ionization_rate = &HeII_ir;
+    grackle_fields.RT_H2_dissociation_rate = &H2_dr;
+    grackle_fields.RT_heating_rate         = &hr;
+    gr_float isrf                          = 1.7;
+    grackle_fields.isrf_habing             = &isrf;
```

Comment:
Do these configuration fields stay the same for all particles and even between iterations?